### ISSUE TITLE
clean exit on clean exit

### ISSUE THREAD

**Issue by iccicci** [remy/nodemon#1719]
Hi all,

default behavior of nodemon when launched program correctly exits it to wait for a file change in order to restart the program.

What I need is to make nodemon to exit when the launched program exits.
I was able to achieve this using the `--exitcrash` nodemon command line option plus a `process.exit(1)` in my launched program; the minor issue I have is that nodemon exits with an error as well.

Is there a way make nodemon exit with `0` exit code when launched program does the same?
If not, would it be possible to add a dedicated command line option?

Thank you,
iCC

**Comment by remy** [comment]
Hi, I'm "all".

As per the [design principles](https://github.com/remy/nodemon#design-principles) I'd rather not introduce new options unless they have a common and real use case.

Which leads me to the obvious question: _why_?

**Comment by iccicci** [comment]
Hi "all" ( AKA @remy 😄 ),

for a customer, I'm working on a complex configuration project:
under windows terminal I have to launch ant, which launches "npm start", which launches concurrently, which (obviously concurrently) launches the client (a create-react-app) and nodemon which launches the express server.

Now, in this hard configuration I can't change (as it is a customer requirement) hitting `<CTRL> + C` on the terminal leaves some node processes alive (which is bad as they can keep opened listening ports or opened log files) which have bad impact on next launches of dev application.

At the end I had to add a "close" button in my react app (obviously visible and active only in local dev env), which calls a dedicated API (obviously visible and bla bla as before) on the server to shutdown.
With `--kill-others` option of concurrently and the stuff I described before in nodemon and server I was able to achieve a gracefull shutdown of local dev app, but ant output is an error output since `the process.exit(1)` from by server is propagated to nodemon then to concurrently than to npm then to ant.

I hope I was able to explain the problem and moreover the scenario.

Anyway, thank you very much for you quick replay and your time, I appreciate so much.

iCC

**Comment by iccicci** [comment]
Hi @remy ,

I'm back to you with some news.
Actually `--exitcrash` has bad collateral effect: when developing I write a syntax error, it causes my entire app to exit with error 😞 
The perfect solution for me should be using a new option which makes nodemon exiting on 0 exit code but not on failure exit code.

Thank you,
iCC

**Comment by remy** [comment]
Sure, but I don't understand how you're using nodemon, a dev tool, which is
why I asked, why.

Nodemon is designed to keep restarting your process during development. How
are you trying to use it?

**Comment by iccicci** [comment]
Hi @remy ,

that way! The scenario I described you is the development setup of my customer.
For deploy we use docker etc etc etc

As per customer request, the development of all pieces of the entire project is done through ant, also the step to launch my react+express piece.

I know concurrectly + nodemon + create-react-app is what we need to develop a react+express project, under Linux this works greatly; but to respect the entire project standard I had to embed this in an ant run, plus I have to this under Windows (platform where I'm not so strong) the result is that when I need to stop the dev server (i.e. to install a new dependency) simply hitting `<CTRL> + C` can have the bad effect I described.

I hope now I was more clear.

To recap my request: a new option which acts as `--exitcrash` **but only** when launched program exits with a `0` exit code.

It would me a lot.

Thank you once again,
iCC

**Comment by remy** [comment]
If I'm honest I still don't quite follow your use case and I'm pretty strongly against adding a new option. However, it stands to reason that the `--exitcrash` should _successfully_ exit nodemon (with exit code `0`) since it was user configured that way.

Which suggests it should help you but also clarify implicit usage.

**Comment by iccicci** [comment]
Hi @remy ,

didn't know that. Now I've worked a while on the project I realized it would not be enough anyway.

First of all let me try to recap my env and use case.
My customer project is a set of  subprojects (one of them is my react+express project).
As per design, my customer wants every subproject to be nested in an ant flow, so to edit/change/test my subproject I need to launch it with an ant command (which launches concurrently, which launches nodemon), all this in a Windows Power Shell.

As I said before, after some days on the project now I'm understanding a bit more on my env and probably the change I'm requesting here is not the best solution for my problem.

I'll try to solve my problem acting on concurrently; in the mean while I close this issue; if I'll not be able to find a solution thanks to concurrently, I'll open another (more detailed) new feature request for nodemon.

Thank you for your time,
iCC
